bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/gpi/GpiCB.java

```diff
@@ -44,46 +44,50 @@ protected void thisReset(Map<String, Object> options) {
         final String systemName = _sysConfig.getSystemName();
-        final Collection<IParameter> sysConfig = _sysConfig.getParameters();
+        Collection<IParameter> sysConfig = _sysConfig.getParameters();
+
+        final class MagnitudeFilter implements Function1<Magnitude, Boolean> {
+            private final MagnitudeBand band;
+
+            private MagnitudeFilter(MagnitudeBand band) {
+                this.band = band;
+            }
+
+            @Override
+            public Boolean apply(Magnitude magnitude) {
+                return magnitude.band().equals(band);
+            }
+        }
 
-        for (final IParameter param : sysConfig) {
+        final class MagnitudeSetter implements ApplyOp<Magnitude> {
+            private final String magProp;
+
+            private MagnitudeSetter(String magHProp) {
+                this.magProp = magHProp;
+            }
+
+            @Override
+            public void apply(Magnitude magnitude) {
+                config.putParameter(systemName,
+                            DefaultParameter.getInstance(magProp,
+                                magnitude.value()));
+            }
+        }
+        for (IParameter param : sysConfig) {
             config.putParameter(systemName,
                     DefaultParameter.getInstance(param.getName(), param.getValue()));
         }
         config.putParameter(systemName,
-                StringParameter.getInstance(InstConstants.INSTRUMENT_NAME_PROP,
-                        Gpi.INSTRUMENT_NAME_PROP));
-
-        // Set the magnitude properties.
-        final ISPObsComponent targetcomp = SPTreeUtil.findTargetEnvNode(getObsComponent().getContextObservation());
-        if (targetcomp != null) {
-            final TargetObsComp toc = (TargetObsComp) targetcomp.getDataObject();
-            if (toc != null) {
-                // The Asterism here should be Single(base), so we only need to concern ourselves with the head target.
-                toc.getAsterism().allSpTargetsJava().headOption().foreach(base -> {
-                    final ImList<Magnitude> magnitudes = base.getMagnitudesJava();
-                    MAG_PROPERTIES.forEach((band, propName) ->
-                        magnitudes.find(m -> m.band().equals(band)).
-                                foreach(mag -> config.putParameter(systemName, DefaultParameter.getInstance(propName, mag.value()))));
-                });
-            }
-        }
+                            StringParameter.getInstance(InstConstants.INSTRUMENT_NAME_PROP,
+                                    Gpi.INSTRUMENT_NAME_PROP));
     }
 
-    // The magnitude bands of interest and their property names in the configuration.
-    private static final Map<MagnitudeBand, String> MAG_PROPERTIES = new HashMap<>();
-    static {
-        MAG_PROPERTIES.put(MagnitudeBand.H$.MODULE$, Gpi.MAG_H_PROP);
-        MAG_PROPERTIES.put(MagnitudeBand.I$.MODULE$, Gpi.MAG_I_PROP);
-    }
-}
```

Comment:
I'm confused about these changes to `GpiCB`.  It doesn't appear in `release/ocs-2019B` or in `develop` and seems unrelated to NGS2.  I'll put all this back like it was unless I hear otherwise.

Reply:
I can't see how this in NGS2 releated. GPI doesn't use AGS at all. I'd remove it

Reply:
Reset to original.